library/rego/sbom-validation.rego

```diff
@@ -0,0 +1,63 @@
+
+# Copyright The Ratify Authors.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+package ratify.policy
+
+# This template defines policy for SBOM validation.
+# It checks the following:
+# - If there are any system errors
+# - If there are errors for any of the images
+# - There is at least one SBOM report that was verified
+# - Only considers ONE SBOM report
+# - The SBOM is valid (isSuccess = true)
+# - The SBOM has a valid notary project signature (if require_signature = true)s
+
+import future.keywords.if
+import future.keywords.in
+
+default require_signature := false # change to true to require notary project signature on vulnerability report
+default valid := false
+
+valid {
+    not failed_verify(input)
+}
+
+failed_verify(reports) {
+    not process_sboms(reports)
+}
+
+process_sboms(subject_result) if {
+    # collect verifier reports from sbom verifier
+    sbom_results := [res | subject_result.verifierReports[i].verifierReports[j].name == "sbom"; res := subject_result.verifierReports[i].verifierReports[j]]
```

Comment:
should we check `type` or `name` for `sbom` verifier`?

Reply:
Looks like the sample had checks the "type". Could you confirm when would customer apply library/rego , vs samples/template.?

Reply:
Yes you're right. It should be using `type`. The vuln report rego in the `library/rego` folder should be updated too. The `library/rego` folder should ONLY be used with rego policy provider.

Now, there's currently an issue with CLI mode with rego policy provider. Current config.json does not include `type` field in the verifier config section. And the plugin verifier passes the raw config to the plugins during execution [here](https://github.com/deislabs/ratify/blob/a25c79107478576df49faf6762dc1970538ec9be/pkg/verifier/plugin/plugin.go#L136). This means plugins will not have the `type` field unless explicitly specified in config.json. Thus, the verifier reports returned from plugins will NOT have the `type` field only have the `name` field. 

The current workaround is adding the type to the json and it works:

```json
"verifier": {
        "version": "1.0.0",
        "plugins": [
            {
                "name": "vulnerabilityreport",
                "type": "vulnerabilityreport",
                "artifactTypes": "application/sarif+json",
                "disallowedSeverities": ["critical"]
            }
        ]
    }
```
Is this the intended behavior moving forward or for CLI scenarios should we add some extra handling to automatically add the type field to the config? cc: @junczhu

Reply:
I remember for backward compatibility, if users don't provide `type` in verifier config, verifier report will skip the `type` to keep the behavior unchanged.

Reply:
Yes that's correct. But I think we should still append the `type` field to the returning verifier report. This way we don't need to manage multiple versions of the rego templates. IMO I don't think this would be a breaking change since the existing `name` field would still be in the report and instead the `type` would just be added as a duplicate value. purely additive. It all should be backwards compatible still. I've added changes to all the verifier to set the `type` field in the report to default to `name` field if the plugin input config doesn't have the `type` field already specified. Please let me know if you think we should handle it differently.